tests/run/float.rs

```diff
@@ -0,0 +1,31 @@
+// Compiler:
+//
+// Run-time:
+//   status: 0
+
+#![feature(const_black_box)]
+
+/*
+ * Code
+ */
+
+fn main() {
+    use std::hint::black_box;
+
+    macro_rules! check {
+        ($ty:ty, $expr:expr) => {
+            {
+                const EXPECTED: $ty = $expr;
+                assert_eq!($expr, EXPECTED);
+            }
+        };
+    }
+
+    check!(i32, (black_box(0.0f32) as i32));
+
+    check!(i64, (black_box(f64::NAN) as i64));
+    check!(u64, (black_box(f64::NAN) as u64));
```

Comment:
Does this also work for `black_box(f32::NAN) as u64/u128`, which is what is mentionned in #75?

Reply:
it does, I've added those tests just to make sure.